sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientJavaDocCodeSnippets.java

```diff
@@ -228,39 +228,40 @@ public void beginRecognizeContentWithOptions() throws IOException {
     /**
      * Code snippet for {@link FormRecognizerAsyncClient#beginRecognizeReceiptsFromUrl(String)}
      */
+    @SuppressWarnings("unchecked")
     public void beginRecognizeReceiptsFromUrl() {
         // BEGIN: com.azure.ai.formrecognizer.FormRecognizerAsyncClient.beginRecognizeReceiptsFromUrl#string
         String receiptUrl = "{file_source_url}";
         formRecognizerAsyncClient.beginRecognizeReceiptsFromUrl(receiptUrl).subscribe(recognizePollingOperation -> {
             // if training polling operation completed, retrieve the final result.
             recognizePollingOperation.getFinalResult().subscribe(recognizedReceipts -> {
                 for (int i = 0; i < recognizedReceipts.size(); i++) {
-                    RecognizedReceipt recognizedReceipt = recognizedReceipts.get(i);
-                    Map<String, FormField> recognizedFields = recognizedReceipt.getRecognizedForm().getFields();
+                    RecognizedForm recognizedReceipt = recognizedReceipts.get(i);
+                    Map<String, FormField<?>> recognizedFields = recognizedReceipt.getFields();
                     System.out.printf("----------- Recognized Receipt page %s -----------%n", i);
-                    FormField merchantNameField = recognizedFields.get("MerchantName");
-                    if (merchantNameField.getFieldValue().getType() == FieldValueType.STRING) {
+                    FormField<?> merchantNameField = recognizedFields.get("MerchantName");
+                    if (merchantNameField.getFieldValue() instanceof String) {
                         System.out.printf("Merchant Name: %s, confidence: %.2f%n",
-                            merchantNameField.getFieldValue().asString(),
+                            merchantNameField.getFieldValue(),
                             merchantNameField.getConfidence());
                     }
-                    FormField transactionDateField = recognizedFields.get("TransactionDate");
-                    if (transactionDateField.getFieldValue().getType() == FieldValueType.DATE) {
+                    FormField<?> transactionDateField = recognizedFields.get("TransactionDate");
+                    if (transactionDateField.getFieldValue() instanceof LocalDate) {
                         System.out.printf("Transaction Date: %s, confidence: %.2f%n",
-                            transactionDateField.getFieldValue().asDate(),
+                            transactionDateField.getFieldValue(),
                             transactionDateField.getConfidence());
                     }
```

Comment:
This is an example where the check for `instanceof` made no difference to the code within the `if` block. So, showing the cast to `LocalDate` and using some fields of `LocalDate` object would be a more effective example.